test/test-gelato-tasks.js

```diff
@@ -240,24 +246,69 @@ describe("Test Gelato Task Creation / Execution", function () {
     // The rest of the payout was transferred to the policy holder
     await expect(tx).to.changeTokenBalance(currency, cust, _A("999.992491") /* $1000 payout - $0.007509 fee */);
 
-    // TODO: Task should be cancelled after this
+    // The task was removed from gelato
+    await expect(tx).to.emit(automate, "TaskCancelled").withArgs(taskCreatedEvent.args.taskId, fpa.address);
   });
-});
 
-// TODO: task cancelation on expiration
+  it("Removes task on expiration", async () => {
+    const { pool, fpa, rm, cust, currency, automate } = await helpers.loadFixture(deployPoolFixture);
+
+    await currency.connect(cust).approve(fpa.address, _A(2000));
+
+    const start = await helpers.time.latest();
+
+    // Create a new policy that expires in 24 hours
+    const creationTx = await fpa
+      .connect(cust)
+      .newPolicy(rm.address, _W("0.57"), true, _A(1000), start + HOUR * 24, cust.address);
+    const taskCreatedEvent = await getTransactionEvent(automate.interface, await creationTx.wait(), "TaskCreated");
+
+    // Policy expires
+    await helpers.time.increase(HOUR * 24);
+    const policy = (await rm.getPolicyData(makePolicyId(rm.address, 1)))[0];
+    const tx = await pool.expirePolicy(policy);
+
+    // The task has been cancelled
+    await expect(tx).to.emit(automate, "TaskCancelled").withArgs(taskCreatedEvent.args.taskId, fpa.address);
+
+    // No funds were transferred to or from the customer
+    await expect(tx).to.changeTokenBalance(currency, cust, 0);
+
+    // No funds were transferred to or from the payout automation contract
+    await expect(tx).to.changeTokenBalance(currency, fpa, 0);
+    await expect(tx).to.changeEtherBalance(fpa, 0);
+  });
+
+  it.only("Gives infinite allowance to the swap router on initialization", async () => {
```

Comment:
Ojo, te quedó un `it.only` !!! 

Tal vez deberíamos tener un pre-commit check que nos evite esto . Ya me pasó otras veces.

Reply:
Otra cosa que podríamos ver es incorporar un plugin similar al que usamos en off-chain que nos control el coverage y que reporte el cambio de coverage, y que nos genere un badge.